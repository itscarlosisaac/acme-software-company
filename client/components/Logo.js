import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types';
export default class Logo extends Component {
  render() {
    return (
      <Fragment>
        <img className="app__logo" src="./assets/logo.svg" width={this.props.w} />
      </Fragment>
    )
  }
}

Logo.propTypes = {
  w: PropTypes.number,
}

Logo.defaultProp = {
  w: 140,
};