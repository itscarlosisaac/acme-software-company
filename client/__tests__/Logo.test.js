import React from 'react';
import Logo from '../components/Logo';

import { configure, shallow } from 'enzyme';
import Adapter from  'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter() });

describe(`<Logo /> - Logo Component`, () => {

  it(`Renders the component`, () => {
    const wrapper = shallow(<Logo w={140} />);
    expect(wrapper.exists()).toBe(true);
  });

  it(`Should accept a size prop`, () => {
    const wrapper = shallow(<Logo w={140} />);
    expect(wrapper.instance().props.w).toBe(140);
  });

  it(`Should render an image tag`, () => {
    const wrapper = shallow(<Logo w={140} />);
    expect(wrapper.find('img').type()).toEqual('img')
  });

  it(`Should Match the snapshot`, () => {
    const tree = shallow(<Logo w={140} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});