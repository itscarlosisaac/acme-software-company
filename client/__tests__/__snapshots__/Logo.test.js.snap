// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> - Logo Component Should Match the snapshot 1`] = `
<Fragment>
  <img
    className="app__logo"
    src="./assets/logo.svg"
    width={140}
  />
</Fragment>
`;
